src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -37,7 +37,7 @@ public IEnumerable<LockFileTargetLibrary> GetRuntimeLibraries(IEnumerable<string
             Dictionary<string, LockFileTargetLibrary> libraryLookup =
                 runtimeLibraries.ToDictionary(e => e.Name, StringComparer.OrdinalIgnoreCase);
 
-            HashSet<string> allExclusionList = new HashSet<string>();
+            HashSet<string> allExclusionList = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Doing a quick code search, it appears

https://github.com/dotnet/sdk/blob/c663c1911c2d96a95c4433cc2900a7e8652b1005/src/Tasks/Microsoft.NET.Build.Tasks/LockFileExtensions.cs#L85

has the same issue.  What do you think about changing that as well?  That method is only called from this method and unioned with this hashset, but in case someone else adds a call to it.